src/Glpi/Form/ServiceCatalog/ServiceCatalogManager.php

```diff
@@ -35,110 +35,220 @@
 
 namespace Glpi\Form\ServiceCatalog;
 
-use Glpi\Form\AccessControl\FormAccessControlManager;
-use Glpi\Form\AccessControl\FormAccessParameters;
-use Glpi\Form\Form;
-use Glpi\FuzzyMatcher\FuzzyMatcher;
-use Glpi\FuzzyMatcher\PartialMatchStrategy;
+use Glpi\Form\ServiceCatalog\Provider\CategoryProvider;
+use Glpi\Form\ServiceCatalog\Provider\CompositeProviderInterface;
+use Glpi\Form\ServiceCatalog\Provider\FormProvider;
+use Glpi\Form\ServiceCatalog\Provider\LeafProviderInterface;
 
 final class ServiceCatalogManager
 {
-    private FormAccessControlManager $access_manager;
-    private FuzzyMatcher $matcher;
+    /** @var \Glpi\Form\ServiceCatalog\Provider\ItemProviderInterface[] */
+    private array $providers;
 
     public function __construct()
     {
-        $this->access_manager = FormAccessControlManager::getInstance();
-        $this->matcher = new FuzzyMatcher(new PartialMatchStrategy());
+        $this->providers = [
+            new FormProvider(),
+            new CategoryProvider(),
+        ];
     }
 
     /**
-     * Return all available forms for the given user.
+     * Return all available forms and non empties categories for the given user.
      *
-     * Forms names may be altered to ensure uniqueness.
-     * This is done by by adding suffixes to forms with the same name :
-     * "My form", "My form (1)", "My form (2)", ...
```

Comment:
This feature has been removed as it is not very important and it is hard to maintain properly here.
If we really want to keep something like this it probably need to be done using some custom twig filter instead.